src/Pdo/PdoInsertCapableTrait.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use PDOStatement;
+
+/**
+ * Common functionality for objects that can insert records into a database using PDO.
+ *
+ * @since [*next-version*]
+ */
+trait PdoInsertCapableTrait
+{
+    /**
+     * Executes an INSERT SQL query, inserting records into the database.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $records A list of containers, each containing the record data to insert.
```

Comment:
Don't know why not, it's a good idea. I'll make the changes.

Reply:
After giving it some more thought, it makes sense to be able to accept a traversable of containers here, as you suggested [here](https://github.com/RebelCode/cqrs-resource-model-abstract/pull/1#discussion_r158841594).

Not only does this make sense given how we view container, but it would also allow us to pass data models directly without having to transform them.

Reply:
I would say we pick something and stick to it, at least for each record, i.e. an array of traversable of _either_ sub-arrays or containers.

At this point, the only thing that's making me hesitant from using containers is the fact that consumers are then bound to using them which might mean they would have to construct them. Is this a valid concern?

Reply:
@XedinUnknown seems to have identified the same problem while working on something else, and has come up with a solution which I quite like. It involves depending on abstract methods which functional similar to a container's `get()` and `has()` methods, except that they have a container/array as parameter.

This means that our classes can use these methods to retrieve something by key while not knowing what they are retrieving from, be it a container, array or something else. But most importantly, it allows us to throw container-related exceptions for non-container collections, such as arrays. With traits that implement this functionality, this seems to be the best approach.

So I will make the resource model abstracts accept both arrays and container, and depend on these abstract methods for internal reading.